diff -ur autobahn-0.10.9.orig/autobahn/twisted/wamp.py autobahn-0.10.9/autobahn/twisted/wamp.py
--- autobahn-0.10.9.orig/autobahn/twisted/wamp.py	2015-09-05 02:52:06.000000000 +0200
+++ autobahn-0.10.9/autobahn/twisted/wamp.py	2015-10-23 16:00:59.208667552 +0200
@@ -549,7 +549,7 @@
         """
         factory = WampWebSocketClientFactory
 
-        def __init__(self, url, realm, make, extra=None,
+        def __init__(self, url, realm, make, context_factory=None, extra=None,
                      debug=False, debug_wamp=False, debug_app=False):
             """
 
@@ -574,6 +574,7 @@
             """
             self.url = url
             self.realm = realm
+            self.context_factory = context_factory
             self.extra = extra or dict()
             self.debug = debug
             self.debug_wamp = debug_wamp
@@ -603,10 +604,12 @@
 
             if isSecure:
                 from twisted.application.internet import SSLClient
-                clientClass = SSLClient
+                if self.context_factory is None:
+                    from twisted.internet import ssl
+                    self.context_factory = ssl.ClientContextFactory()
+                client = SSLClient(host, port, transport_factory, self.context_factory)
             else:
                 from twisted.application.internet import TCPClient
-                clientClass = TCPClient
+                client = TCPClient(host, port, transport_factory)
 
-            client = clientClass(host, port, transport_factory)
             client.setServiceParent(self)
